### ISSUE TITLE
File of new and changed gene names

### ISSUE THREAD

**Issue by gw3** [WormBase/website#4679]
There was a discussion recently about having a display on the website of some selected genes which have been recently assigned CGC names.

To facilitate this, there is now a new file in the database release FTP site detailing changes in gene names. This describes existing genes whose gene name has changed, new gene names that have been assigned, gene names that have been retired, gene names that have been assigned to another gene, and genes with a gene name whose Biotype has changed.

See:
ftp://ftp.sanger.ac.uk/pub/wormbase/releases/WS253/species/c_elegans/PRJNA13758/annotation/c_elegans.PRJNA13758.WS253.changed_CGC_names.txt

**Comment by kyook** [comment]
Thanks Gary, this file is great.  A couple questions though, 
What is happening in the  # removed CGC names group.  Are these just killed genes? 
# old CGC	old GeneID	old SequenceID	old Biotype	new CGC	new GeneID	new SequenceID	new Biotype
wdr-81	WBGene00201617	F52C9.10	Transcript	.	.	.	.
nhr-284	WBGene00020606	T20C7.2	CDS	.	.	.	.

Also, I'm confused about the  # assigned Gene changed group.  How is this different from the changed CGC name group?

**Comment by gw3** [comment]
Hi Karen,

In the 'removed CGC names' group, the CGC name that was present in the previous database release is not present in the current database release. All the script is reporting is that these CGC_names are no longer assigned as the CGC_name of any genes. The script that makes this file does not investigate whether  the gene it was assigned to in the previous release still exists or not. 

I could add information on whether the gene still exists if it is useful, but I think taking a minute to check things out would be more informative. For example, looking in WS253 at the Remarks in WBGene00201617, we find:
"wdr-81 erroneously connected to this gene in WB251 and WS251. wdr-81 is WBGene00018672"
Looking at WBGene00018672, we can confirm that 'wdr-81' is now an Other_name of WBGene00018672.

.... and for the missing name 'nhr-284', the gene it was on (WBGene00020606) has been merged into WBGene00015897. The gene WBGene00015897 has the CGC name 'nhr-156', with 'nhr-284' now being its Other_name.

I think trying to summarise these stories in a simple table could get messy, but I'm open to suggestions.



For the group 'assigned Gene changed'. This is the set of CGC_names which exist in the previous database release on one gene, but now they have been assigned as a CGC_name to a different gene. I think that this would be a very rare event as when we do gene merges or gene splits, we try hard to keep the CGC_name associated with its old WBGeneID object. I think part of your confusion here is understandable because all of the groups except for the first one (Genes with changed CGC names) are looking at the CGC_name and reporting differences in the properties (WBGeneID, Biotype, Sequence_name_ID) of the gene it it linked to. 
In the first group, however, (Genes with changed CGC names), I was trying to be helpful and linking up two different CGC_names, one of which was linked to a gene in the previous database release and another one which is linked to the same gene in the current database release. So the first group is looking at CGC_name changes from the viewpoint of a gene, rather than that of a CGC_name.

So the group 'Genes with changed CGC names', is those genes which have changed their CGC_names.
The group 'assigned Gene changed' is those CGC_names which have changed their genes.

**Comment by kyook** [comment]
Hi Gary,

Thanks, but I'm still having trouble understanding what this means:

"looking in WS253 at the Remarks in WBGene00201617, we find:
"wdr-81 erroneously connected to this gene in WB251 and WS251. wdr-81 is
WBGene00018672"
Looking at WBGene00018672, we can confirm that 'wdr-81' is now an
Other_name of WBGene00018672."

So wdr-81 was erroneously connected to WBGene00201617 in WS251, and is
actually WBGene00018672, which are both F52C9.1? Is this a merger of two
WBGeneID's and now there is no CGC public name associated with either id?
So then in this case both WBGene IDs are still valid, but they no longer
have an official CGC gene name?

Given the gene name changes on the ftp site, I think we can come up with a
simple way for displaying the changes. I think there should be one table
for the newly named genes and one table for all the renames, removed,
changed, merged etc. genes. I've pasted examples below, and attached a text
file, in case that is easier to read.

For the newly named loci, the table can have headers and layout such as:
WBGeneID Sequence Sequence type WS253 Gene public name Laboratory/Reference
WBGene00016879 C52D10.12 CDS cnnm-1 T. Ishii and H.Miki
WBGene00016283 C31B8.8 CDS zmp-3 Altincicek et al. (WBPaper00036427)
WBGene00007255 C01H6.6 CDS cnnm-4 T. Ishii and H.Miki
WBGene00045388 C38C6.8 Transcript lfor-1
WBGene00019198 H14E04.1 CDS strm-1 T. Kurzchalia
WBGene00045161 C38C6.7 Transcript lfor-2
WBGene00020646 T21D11.1 CDS zmp-6 Altincicek et al. (WBPaper00036427)
WBGene00019212 H19M22.3 CDS zmp-2 Altincicek et al. (WBPaper00036427)
WBGene00016343 C33D12.2 CDS cnnm-3 T. Ishii and H.Miki
WBGene00012364 W09D12.1 CDS zmp-4 Altincicek et al. (WBPaper00036427)
WBGene00017816 F26B1.2 CDS hrpk-1 A.Y. Zinovyeva
WBGene00019869 R04E5.2 CDS cnnm-2 T. Ishii and H.Miki
WBGene00010423 H36L18.1 CDS zmp-5 Altincicek et al. (WBPaper00036427)
WBGene00011260 R13G10.4 CDS cnnm-5 T. Ishii and H.Miki

For the changed gene names table the columns should be a on the past and
current release, with a comment column for the curator remark.
For example
WS252 GeneID WS252 gene public name WS252 SequenceID WS252 gene type WS253
GeneID WS253 gene public name WS253 SequenceID WS253 gene type Comment
WBGene00016737 miro-2 C47C12.4 CDS WBGene00016737 miro-2 C47C12.4
Pseudogene Gene
type changed
WBGene00007667 fbxa-97 C18B12.5 CDS WBGene00007667 fbxa-97 C18B12.5
Pseudogene Gene type changed
WBGene00015802 flu-2 C15H9.7 CDS WBGene00015802 kynu-1 C15H9.7 CDS Public
gene name changed
WBGene00007050 tag-184 F18C5.3 CDS WBGene00007050 utp-20 F18C5.3 CDS Public
gene name changed
WBGene00201617 wdr-81 F52C9.10 Transcript . . . . Public gene name removed
- wdr-81 erroneously connected to this gene. wdr-81 is correctly connected
to WBGene00018672
WBGene00020606 nhr-284 T20C7.2 CDS . . . . Public gene name removed:
WBGene00020606 has been merged into WBGene00015897, with public name
'nhr-156'; 'nhr-284' now an Other_name for WBGene00015897.
~~~~
Karen Yook

Curator
WormBase Caltech
Tel: 415.306.4150
e-mail: kyook@caltech.edu
e-mail: karen@wormbase.org
skype name: wbkaren


On Fri, Mar 18, 2016 at 3:17 AM, gw3 <notifications@github.com> wrote:

> Hi Karen,
>
> In the 'removed CGC names' group, the CGC name that was present in the
> previous database release is not present in the current database release.
> All the script is reporting is that these CGC_names are no longer assigned
> as the CGC_name of any genes. The script that makes this file does not
> investigate whether the gene it was assigned to in the previous release
> still exists or not.
>
> I could add information on whether the gene still exists if it is useful,
> but I think taking a minute to check things out would be more informative.
> For example, looking in WS253 at the Remarks in WBGene00201617, we find:
> "wdr-81 erroneously connected to this gene in WB251 and WS251. wdr-81 is
> WBGene00018672"
> Looking at WBGene00018672, we can confirm that 'wdr-81' is now an
> Other_name of WBGene00018672.
>
> .... and for the missing name 'nhr-284', the gene it was on
> (WBGene00020606) has been merged into WBGene00015897. The gene
> WBGene00015897 has the CGC name 'nhr-156', with 'nhr-284' now being its
> Other_name.
>
> I think trying to summarise these stories in a simple table could get
> messy, but I'm open to suggestions.
>
> For the group 'assigned Gene changed'. This is the set of CGC_names which
> exist in the previous database release on one gene, but now they have been
> assigned as a CGC_name to a different gene. I think that this would be a
> very rare event as when we do gene merges or gene splits, we try hard to
> keep the CGC_name associated with its old WBGeneID object. I think part of
> your confusion here is understandable because all of the groups except for
> the first one (Genes with changed CGC names) are looking at the CGC_name
> and reporting differences in the properties (WBGeneID, Biotype,
> Sequence_name_ID) of the gene it it linked to.
> In the first group, however, (Genes with changed CGC names), I was trying
> to be helpful and linking up two different CGC_names, one of which was
> linked to a gene in the previous database release and another one which is
> linked to the same gene in the current database release. So the first group
> is looking at CGC_name changes from the viewpoint of a gene, rather than
> that of a CGC_name.
>
> So the group 'Genes with changed CGC names', is those genes which have
> changed their CGC_names.
> The group 'assigned Gene changed' is those CGC_names which have changed
> their genes.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/WormBase/website/issues/4679#issuecomment-198293449>
>

WBGeneID	Sequence	Sequence type	WS253 Gene public name	Laboratory/Reference
WBGene00016879	C52D10.12	CDS	cnnm-1	T. Ishii and H.Miki
WBGene00016283	C31B8.8	CDS	zmp-3	Altincicek et al. (WBPaper00036427)
WBGene00007255	C01H6.6	CDS	cnnm-4	T. Ishii and H.Miki
WBGene00045388	C38C6.8	Transcript	lfor-1
WBGene00019198	H14E04.1	CDS	strm-1	T. Kurzchalia
WBGene00045161	C38C6.7	Transcript	lfor-2
WBGene00020646	T21D11.1	CDS	zmp-6	Altincicek et al. (WBPaper00036427)
WBGene00019212	H19M22.3	CDS	zmp-2	Altincicek et al. (WBPaper00036427)
WBGene00016343	C33D12.2	CDS	cnnm-3	T. Ishii and H.Miki
WBGene00012364	W09D12.1	CDS	zmp-4	Altincicek et al. (WBPaper00036427)
WBGene00017816	F26B1.2	CDS	hrpk-1	A.Y. Zinovyeva
WBGene00019869	R04E5.2	CDS	cnnm-2	T. Ishii and H.Miki
WBGene00010423	H36L18.1	CDS	zmp-5	Altincicek et al. (WBPaper00036427)
WBGene00011260	R13G10.4	CDS	cnnm-5	T. Ishii and H.Miki




WS252 GeneID	WS252 gene public name	WS252 SequenceID	WS252 gene type	WS253 GeneID	WS253 gene public name	WS253 SequenceID	WS253 gene type	Comment	Reference
WBGene00016737	miro-2	C47C12.4	CDS	WBGene00016737	miro-2	C47C12.4	Pseudogene	Gene type changed
WBGene00007667	fbxa-97	C18B12.5	CDS	WBGene00007667	fbxa-97	C18B12.5	Pseudogene	Gene type changed
WBGene00015802	flu-2	C15H9.7	CDS	WBGene00015802	kynu-1	C15H9.7	CDS	Public gene name changed
WBGene00007050	tag-184	F18C5.3	CDS	WBGene00007050	utp-20	F18C5.3	CDS	Public gene name changed
WBGene00201617	wdr-81	F52C9.10	Transcript	.	.	.	.	Public gene name removed - wdr-81 erroneously connected to this gene. wdr-81 is correctly connected to WBGene00018672
WBGene00020606	nhr-284	T20C7.2	CDS	.	.	.	.	Public gene name removed:  WBGene00020606 has been merged into WBGene00015897, with public name 'nhr-156'; 'nhr-284' now an Other_name for WBGene00015897.

**Comment by tuli** [comment]
@kyook let me clear up the first issue you raise.

Gary wrote: 
"looking in WS253 at the Remarks in WBGene00201617, we find:
"wdr-81 erroneously connected to this gene in WB251 and WS251. wdr-81 is
WBGene00018672"
Looking at WBGene00018672, we can confirm that 'wdr-81' is now an
Other_name of WBGene00018672."

you replied: 
So wdr-81 was erroneously connected to WBGene00201617 in WS251, and is
actually WBGene00018672, which are both F52C9.1? Is this a merger of two
WBGeneID's and now there is no CGC public name associated with either id?
So then in this case both WBGene IDs are still valid, but they no longer
have an official CGC gene name?

My response:
wdr-81 was erroneously connected to WBGene00201617 in WS250 and WS251 (I've corrected the typo in the Remark for WS254). WBGene00201617 is F52C9.10 

wdr-81 should have been assigned to WBGene00018672 which is F52C9.1. 
Meanwhile, WBGene00018672 was assigned sorf-2 in WS253 and wdr-81 became an other_name. 

So, wdr-81 was moved from WBGene00201617 to an other name of WBGene00018672 in WS253.

This is a very unusual situation (mistake, transfer of names, other name, new name) and I think it will be hard to track/document in an automated way - it's why we biocurators haven't been replaced by computers :-)

**Comment by gw3** [comment]
"So wdr-81 was erroneously connected to WBGene00201617 in WS251, and is
actually WBGene00018672, which are both F52C9.1? Is this a merger of two
WBGeneID's and now there is no CGC public name associated with either id?
So then in this case both WBGene IDs are still valid, but they no longer
have an official CGC gene name?"

No, this is not a merger. WBGene00018672 is F52C9.1.  WBGene00201617 is F52C9.10 (the extra '0' on the end is the cause of the original typo that assigned wdr-81 to the wrong gene in the first place!). 
wdr-81 was erroneously connected to WBGene00201617 in WS251. This was corrected in WS253 by moving the name wdr-81 to be an Other_name of WBGene00018672 (which already had been assigned another CGC-name of sorf-2).These sorts of corrections are very rare. I'm sure Mary Ann can fill in the details behind this better than I can.

I don't think that it will be possible to easily cover all descriptions of changes in gene-names.
There will always be rare instances of improvements made to the naming of genes that show up in unusual ways. For instance, in WS250. the name 'yars-1' was moved from WBGene00006968 to WBGene00013677. I had to go back through the genenames mailing list to sort out what was going on: WBGene00006968 had been renamed 'yars-2' and the 'yars-1' name moved to the paralog WBGene00013677 that matched the name better.

It is probably better to ask Mary Ann what has happened when something unusual is found in this  changed_CGC_names.txt file.

I can certainly rearrange the file along the lines you suggest.
Adding comments from curators will be messy though - most Remarks in Gene object have nothing to do with CGC-name assignment. I'll probably add an automatic text description of what appears to be happening in each instance instead.

**Comment by tuli** [comment]
cross-post, but saying the same thing (phew!).

**Comment by gw3** [comment]
Karen,
I've made four example files that I've put in
ftp://ftp.sanger.ac.uk/pub/wormbase/STAFF/gw3/

They are 250.cgc, 251.cgc, 252.cgc and 253.cgc (one for each of the last four releases).
Let me know what you think.

**Comment by kyook** [comment]
Hi Gary,  these files look good to me. 
Thanks!

**Comment by kyook** [comment]
Gary,

Going forward, will you depositing these files in the same directory for
each release? Or perhaps the question is, where can the web team access
these files to produce the webpage announcements?


karen

~~~~
Karen Yook

Curator
WormBase Caltech
Tel: 415.306.4150
e-mail: kyook@caltech.edu
e-mail: karen@wormbase.org
skype name: wbkaren


On Wed, Mar 23, 2016 at 5:04 AM, gw3 <notifications@github.com> wrote:

> Closed #4679 <https://github.com/WormBase/website/issues/4679>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/WormBase/website/issues/4679#event-600177069>
>

**Comment by gw3** [comment]
The changed_CGC_names.txt file will be available in the c_elegans annotation FTP directory that is made with each database relaease. 

So for WS253, it is in:

ftp://ftp.sanger.ac.uk/pub/wormbase/releases/WS253/species/c_elegans/PRJNA13758/annotation/c_elegans.PRJNA13758.WS253.changed_CGC_names.txt
